src/activity/service.test.ts

```diff
@@ -0,0 +1,64 @@
+import { LoggerMock, YagnaMock, agreementPoolServiceMock, paymentServiceMock } from "../../tests/mock";
+import { ActivityPoolService } from "./service";
+import { Activity } from "./activity";
+
+const logger = new LoggerMock();
+const yagnaApi = new YagnaMock().getApi();
+describe("Activity Pool Service", () => {
+  let activityService: ActivityPoolService;
+  beforeEach(() => {
+    activityService = new ActivityPoolService(yagnaApi, agreementPoolServiceMock, paymentServiceMock, {
+      logger,
+    });
+    logger.clear();
+  });
+
+  describe("run()", () => {
+    it("should start service", async () => {
+      await activityService.run();
+      expect(logger.logs).toContain("Activity Pool Service has started");
```

Comment:
I don't like that we're testing the logger here, why not check if `isServiceRunning` is true instead?

Reply:
you are right, but this entire Service is actually a clone of AgreementServicePool, so I duplicated the same structore and tests (and code smells) that are there. but actually in the new file I can do it better now..